crates/rattler_conda_types/src/match_spec/mod.rs

```diff
@@ -146,19 +154,26 @@ impl Display for MatchSpec {
             write!(f, "::")?;
         }
 
-        match &self.name {
-            Some(name) => write!(f, "{name}")?,
-            None => write!(f, "*")?,
+        if let Some(name) = &self.name {
+            write!(f, "{}", name)?;
+        } else {
+            write!(f, "*")?;
         }
 
-        match &self.version {
-            Some(version) => write!(f, " {version}")?,
-            None => (),
+        if let Some(version) = &self.version {
+            write!(f, " {}", version)?;
         }
 
-        match &self.build {
-            Some(build) => write!(f, " {build}")?,
-            None => (),
+        if let Some(build) = &self.build {
+            write!(f, " {}", build)?;
+        }
+
+        if let Some(md5) = &self.md5 {
```

Comment:
This doesn't really produce a proper matchspec that can be parsed again. It would need the brackets I think?